piControlMain.h

```diff
@@ -70,6 +70,13 @@ typedef struct spiControlDev {
 	struct device *dev;
 	struct thermal_zone_device *thermal_zone;
 
+	// supports extension modules with RS485 based communication (eg. DIO or AIO)
+	unsigned int pibridge_supported:1;
+	// device is only equipped on the left side with a PiBridge connector (eg. RevPi Connect)
+	unsigned int only_left_pibridge:1;
+	// device supports RevPi gateways (only point to point communication)
+	unsigned int revpi_gate_supported:1;
+
```

Comment:
Torvalds himself has once ranted that while he finds `bool` acceptable as a function return value, he prefers `unsigned int:1` in struct members. Let me see if I can dig up that message in the LKML archives.

Reply:
Found it: https://lkml.org/lkml/2017/11/21/384

Reply:
Also, coding style:
https://github.com/torvalds/linux/blob/master/Documentation/process/coding-style.rst#17-using-bool

<blockquote>
If a structure has many true/false values, consider consolidating them into a bitfield with 1 bit members, or using an appropriate fixed width type, such as u8.
</blockquote>

Reply:
Interestingly, there was even an attempt to add a check to checkpatch, but it looks like the effort fizzled out after lengthy discussion:
https://patchwork.kernel.org/project/linux-pm/patch/95477c93db187bab6da8a8ba7c57836868446179.camel@perches.com/

The one thing `bool` has got going for is that on some arches, writes can be done atomically with `cmpxchg()` because the size is always (at least) 1 byte. (Though I believe ARM 32-bit is not among them.) That's not possible with a bitfield, which always requires a read-modify-write. However, that's not an issue here as these flags are only set once on driver load and never modified.

Reply:
Thats interesting. Using 1-bit-sized bitfields for boolean expressions still seems weird to me. I do not like too many bool variables in a struct either but my preferred solution would be to implement them as flags by means of bit masks (especially if they can be logically grouped).


> 
> The one thing `bool` has got going for is that on some arches, writes can be done atomically with `cmpxchg()` because the size is always (at least) 1 byte. (Though I believe ARM 32-bit is not among them.) That's not possible with a bitfield, which always requires a read-modify-write. However, that's not an issue here as these flags are only set once on driver load and never modified.

The more serious problem with a bitfields is that the C standard does not specify its layout. Its completely compiler dependent, which is the reason that it should never be used to reference bits in underlying hardware registers.